server/src/main/java/org/jboss/as/server/deployment/module/descriptor/JBossDeploymentStructureParser10.java

```diff
@@ -522,8 +522,22 @@ private static void parseResourceRoot(final DeploymentUnit deploymentUnit, final
                             Closeable closable = null;
                             if(overlay != null) {
                                 overlay.remountAsZip(false);
-                            } else if(child.isFile()) {
-                                closable = VFS.mountZip(child, child, TempFileProviderService.provider());
+                            } else {
+                                if(!child.exists()) {
+                                    // the reference does not exist
+                                    throw ServerLogger.ROOT_LOGGER.illegalResoureRootReference(child.getName());
+                                } else if(child.isFile()) {
+                                    try {
+                                        // check if the resource-root is within the deployment or not
+                                       if(child.getPathNameRelativeTo(deploymentRootFile) != null) {
```

Comment:
@jamezp I agree that the solution regarding the check is a hack but I did not find any other way to rely on VFS to check a path being relative to another path. I mean there is no child.isPathRelativeTo(parent). Maybe @ctomc knows if there is anything in the API regarding this kind of checks.